proposals/generic-ref-struct.md

```diff
@@ -0,0 +1,424 @@
+
+This feature though expands the set of types available. The set of types applicable to `T where T : ref struct` is larger than simply `T`. This means the feature is an anti-constraint yet it occupies a constraint location. 
+
+It is reasonable to consider a new syntax here that cleanly differentiates our anti-constraints from actual constraints. At the time of this writing though there are only two proposed anti-constraints and one is lacking supporting evidence. This makes it questionable if it meets the bar for a new syntax or not.
+
+If a new syntax is desired to separate anti-constraints from constraints then `allow` is one to consider: 
+
+```c#
+readonly ref struct Span<T> 
+    allow T : ref struct
+{
+
+}
+```
+
+This makes the support for `[SupportsOnlyNonByRefLike]` cleaner as the implementation is to simply ignore the set of `allow` items. It may even lead to a better name such as `[RemoveAllows]`. 
+
+The only downside is that it means type parameter declarations get more complex because there are now both `where` and `allow` lists on them.
+
+```c#
+void Write<T>(T value)
+    where T : ISpanFormattable
+    allow T : ref struct
+{
+
+}
+```
+
+At the same time it's possible this will read clearer to developers. It also opens the door for future allows such as pointers. 
+
+### Generalize the delegate support
+The use of any type in `delegate` may cause readers to desire a more general feature that could be applied to any generic parameter. Essentially could we design a syntax for type parameters that could be any type (pointer, `ref`, etc ...) and allow it on any generic parameter? That would leave `delegate` instances as effectively having an implicit version of this syntax.
+
+It is possible to design such a syntax. Let's use `where T : any` for discussion purposes.
+
+```c#
+// Developer writes
+public delegate T Func<T>();
+
+// Developer effectively writes
+public delegate T Func<T>() where T : any;
+```
+
+The problem with this feature is that it's only useful in signature only scenarios. Those work becase the there is no action associated with the generic parameters. The compiler only needs to consider the standard constraint checking. Effectively the compiler needs to ensure that a `T` with `where T : any` isn't passed to an incompatible generic type parameter.
+
+```c#
+interface I1<U> { }
+interface I2<U> where U : IFormattable { }
+
+// Error: T is more permissive than U
+interface IImpl1<T> : I1<T> where T : any { }
+
+// Error: T doesn't implement IFormattable
+interface IImpl2<T> : I2<T> where T : any { }
+```
+
+Once the feature is applied to scenarios where code can execute on `T` it becomes clear that virtually nothing can be done with `T`. That is because the langauge, and runtime, have to consider all the different permutations that `T` can take. Lets look at a few simple examples: 
```

Comment:
```suggestion
Once the feature is applied to scenarios where code can execute on `T` it becomes clear that virtually nothing can be done with `T`. That is because the language, and runtime, have to consider all the different permutations that `T` can take. Lets look at a few simple examples: 
```